servicepulse/troubleshooting.md

```diff
@@ -26,25 +26,37 @@ tags:
-	
+
 When accessed, the web application is working fine. Shortly after accessing the web application, the Heartbeat message is restored and indicates the endpoint status as active.
-	
+
 
 #### Causes and solutions
-	
-The issue is due to the way IIS handles application pools. By default after a certain period of inactivity the application pools is stopped, or, under certain configurable conditions, the application pool is recycled. In both cases the ServicePulse heartbeat is not sent anymore until a new web request comes in waking up the web application.
-	
+
+The issue is due to the way IIS handles application pools. By default after a certain period of inactivity the application pool is stopped, or, under certain configurable conditions, the application pool is recycled. In both cases the ServicePulse heartbeat is not sent anymore until a new web request comes in waking up the web application.
+
 There are two ways to avoid the issue:
-	
+
 1. Configuring IIS to avoid recycling (see possible method [here](http://blogs.msdn.com/b/lucascan/archive/2011/09/30/using-a-windows-azure-startup-script-to-prevent-your-site-from-being-shutdown.aspx));
 2. Use a periodic warm-up HTTP GET to make sure the website is not brought down due to inactivity (the frequency needs to be less than 20 minutes, which is the default IIS recycle-on-idle time)
 
-In some cases configuring IIS to avoid recycling is not possible (for example, when using Azure WebSites or other scenarios in which the IIS is not fully configurable). In these cases, the recommended approach is the second one. It also has the side benefit of avoiding the "first user after idle time" wake-up response-time hit.
+Starting from IIS 7.5 and higher the above steps can be combined into one by following these steps:
+
+1. Enable `AlwaysRunning` mode for the application pool of the site. Go to the application pool management, open the Advanced Settings under General switch the `Start Mode` to `AlwaysRunning`
+1. Enabled Preload for the site itself. Right click on the site, then Manage Site under Advanced Settings in the General settings, switch `Enable Preload` to `true`
+1. Install the Application Initialization Module as described here http://www.iis.net/learn/get-started/whats-new-in-iis-8/iis-80-application-initialization
+1. Add the following to your web.config under the ``<system.webServer>`` node
+
+```
+<applicationInitialization doAppInitAfterRestart="true" >
+<add initializationPage="/" />
+</applicationInitialization>
+```
 
+In some cases configuring IIS to avoid recycling is not possible (for example, when using Azure WebSites or other scenarios in which the IIS is not fully configurable). In these cases, the recommended approach is the second one. It also has the side benefit of avoiding the "first user after idle time" wake-up response-time hit.
```

Comment:
Note: this was written for the old roles. Going to revise this then

Am Dienstag, 4. August 2015 schrieb David Boike :

> In servicepulse/troubleshooting.md
> <https://github.com/Particular/docs.particular.net/pull/728#discussion_r36199443>
> :
>
> >
> > +In some cases configuring IIS to avoid recycling is not possible (for example, when using Azure WebSites or other scenarios in which the IIS is not fully configurable). In these cases, the recommended approach is the second one. It also has the side benefit of avoiding the "first user after idle time" wake-up response-time hit.
>
> I've used that option on low-traffic, high startup time sites and it seems
> to work as advertised.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/docs.particular.net/pull/728/files#r36199443>
> .
>